### PR TITLE
fix: resolve miscellaneous errors in various C examples

### LINKED ISSUE #1690: BUG: running C examples in develop throws an error
### Description

```console
% make examples-c EXAMPLES_FILTER=".*/math/base/special/min/.*"

Running example: /Users/pranavchiku/repos/stdlib-js/stdlib/lib/node_modules/@stdlib/math/base/special/min/examples/c/example.c
Resolving package path...
Package path: /Users/pranavchiku/repos/stdlib-js/stdlib/lib/node_modules/@stdlib/math/base/special/min
Resolving package manifest...
Successfully resolved package manifest.
Resolving include directories...
Resolving source files...
Resolving libraries...
Resolving library paths...
Compiling example...
/Users/pranavchiku/repos/stdlib-js/stdlib/lib/node_modules/@stdlib/math/base/special/min/src/main.c:56:2: warning: no newline at end of file [-Wnewline-eof]
}
 ^
1 warning generated.
example.c:29:25: error: call to undeclared function 'rand'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
        x = ( ( (double)rand() / (double)RAND_MAX ) * 200.0 ) - 100.0;
                        ^
example.c:29:42: error: use of undeclared identifier 'RAND_MAX'
        x = ( ( (double)rand() / (double)RAND_MAX ) * 200.0 ) - 100.0;
                                         ^
example.c:30:42: error: use of undeclared identifier 'RAND_MAX'
        y = ( ( (double)rand() / (double)RAND_MAX ) * 200.0 ) - 100.0;
                                         ^
example.c:34:2: warning: no newline at end of file [-Wnewline-eof]
}
 ^
1 warning and 3 errors generated.
make[1]: *** [example.out] Error 1
Error when attempting to compile example.
ERROR: An error was encountered during execution.

make: *** [examples-c] Error 1
```

```console
% make examples-c EXAMPLES_FILTER=".*/math/base/special/minabs/.*" 

Running example: /Users/pranavchiku/repos/stdlib-js/stdlib/lib/node_modules/@stdlib/math/base/special/minabs/examples/c/example.c
Resolving package path...
Package path: /Users/pranavchiku/repos/stdlib-js/stdlib/lib/node_modules/@stdlib/math/base/special/minabs
Resolving package manifest...
Successfully resolved package manifest.
Resolving include directories...
Resolving source files...
Resolving libraries...
Resolving library paths...
Compiling example...
/Users/pranavchiku/repos/stdlib-js/stdlib/lib/node_modules/@stdlib/math/base/special/minabs/src/main.c:40:2: warning: no newline at end of file [-Wnewline-eof]
}
 ^
1 warning generated.
/Users/pranavchiku/repos/stdlib-js/stdlib/lib/node_modules/@stdlib/math/base/special/min/src/main.c:56:2: warning: no newline at end of file [-Wnewline-eof]
}
 ^
1 warning generated.
example.c:29:25: error: call to undeclared function 'rand'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
        x = ( ( (double)rand() / (double)RAND_MAX ) * 1000.0 ) - 500.0;
                        ^
example.c:29:42: error: use of undeclared identifier 'RAND_MAX'
        x = ( ( (double)rand() / (double)RAND_MAX ) * 1000.0 ) - 500.0;
                                         ^
example.c:30:42: error: use of undeclared identifier 'RAND_MAX'
        y = ( ( (double)rand() / (double)RAND_MAX ) * 1000.0 ) - 500.0;
                                         ^
example.c:34:2: warning: no newline at end of file [-Wnewline-eof]
}
 ^
1 warning and 3 errors generated.
make[1]: *** [example.out] Error 1
Error when attempting to compile example.
ERROR: An error was encountered during execution.

make: *** [examples-c] Error 1
```

### Related Issues

NA.

### Questions

No.

### Demo

_No response_

### Reproduction

```shell
shown above
```

### Expected Results

```shell
examples run successfully
```

### Actual Results

```shell
examples failed to run
```

### Version

latest develop

### Environments

N/A

### Browser Version

_No response_

### Node.js / npm Version

_No response_

### Platform

_No response_

### Checklist

- [X] Read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [X] Searched for existing issues and pull requests.

### PR DESCRIPTION
**Pull request by Pranavchiku**
Resolves #1690.

## Description

> What is the purpose of this pull request?

This pull request resolves failed C examples in develop.

## Related Issues

> Does this pull request have any related issues?

Fixes #1690.

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

File: lib/node_modules/@stdlib/math/base/special/ceiln/src/ceiln.c [hunk 1]
```diff
@@ -32,8 +32,9 @@
 
 // VARIABLES //
 
+#define HUGE 1.0e308
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/ceiln/src/ceiln.c]: What is the advantage of doing this? Why isn't the previous static const declaration sufficient?

File: lib/node_modules/@stdlib/math/base/special/floorn/src/floorn.c [hunk 2]
```diff
@@ -32,8 +32,9 @@
 
 // VARIABLES //
 
+#define HUGE 1.0e308
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/floorn/src/floorn.c]: Same comment as above.

Comment (kgryte) [reply lib/node_modules/@stdlib/math/base/special/floorn/src/floorn.c]: We need to be careful using `define`, as you don't always know what symbols are imported through via `#includes`. You could end up clobbering an existing macro, which is why we'd need to namespace and add include guards. The static const declaration avoids this.

**Review by kgryte — CHANGES_REQUESTED** [review]
@Pranavchiku Left a couple of comments. Not sure about switching to using a macro over a static const.

File: lib/node_modules/@stdlib/math/base/special/ceiln/src/ceiln.c (hunk 1 above)
Comment (Pranavchiku) [reply lib/node_modules/@stdlib/math/base/special/ceiln/src/ceiln.c]: Previous one throws an error

Comment (kgryte) [reply lib/node_modules/@stdlib/math/base/special/ceiln/src/ceiln.c]: The solution then is to rename the original variable, not overwrite the built-in macro. E.g., `static const HUGE_VALUE` or something similar.

Comment (Pranavchiku) [reply lib/node_modules/@stdlib/math/base/special/ceiln/src/ceiln.c]: Yup perfect.

**Review by kgryte — APPROVED** [review]
LGTM. Thanks, @Pranavchiku!

**Comment by Pranavchiku** [comment]
Thanks @kgryte ! Didn't wish to bother you but can you please quickly go through PR #1670?

**Pull request merged**